images_of_bot5.0.py

```diff
@@ -696,40 +696,30 @@ def search_for_places(r):
 
 def swim(r, goodregex, postinto, getfromthese, submission, badregex=r"(\bbadregex\b)", badcaseregex=r"(\bBadCaseRegex\b)", smallsubredditblacklist={'blacklistedsubreddit'}):
         title = submission.title
-        if re.search(goodregex, title, flags=re.IGNORECASE) != None:
-            if re.search(badregex, title, flags=re.IGNORECASE) == None:
-                if re.search(badcaseregex, title,) == None:
-                    if str(submission.author).lower() not in globaluserblacklist:
-                        if str(submission.subreddit).lower() not in globalsubredditblacklist:
-                            if str(submission.subreddit).lower() not in smallsubredditblacklist:
-                                if submission.over_18 == False:
-                                    if re.search(urlregex, submission.url, flags=re.IGNORECASE) != None:
-                                        make_post(r, submission, postinto)
-        else:
-            if str(submission.subreddit).lower() in getfromthese:
-                if str(submission.author).lower() not in globaluserblacklist:
-                
-                    if submission.over_18 == False:
-                        if re.search(urlregex, submission.url, flags=re.IGNORECASE) != None:
-                            make_post(r, submission, postinto)
+        if (submission.over_18  or
+                submission.author.name.lower() in globaluserblacklist or
+                not re.search(urlregex, submission.url, flags=re.IGNORECASE)):
+            return
```

Comment:
\^ if the submission is nsfw, the user is blacklisted, or the url doesn't match the regex, don't bother doing the other more comprehensive checks.